src/Creator.tsx

```diff
@@ -0,0 +1,234 @@
+      kind: 'QuickStarts',
+      ...(selectedTypeMeta?.extraMetadata ?? {}),
+    },
+    spec: {
+      displayName: state.title,
+      icon: null,
+      description: state.description,
+      introduction:
+        selectedTypeMeta?.hasTasks === true ? state.introduction : undefined,
+      prerequisites:
+        selectedTypeMeta?.hasTasks === true ? state.prerequisites : undefined,
+      link:
+        selectedTypeMeta?.fields?.url === true
+          ? {
+              href: state.url,
+              text: 'View documentation',
+            }
+          : undefined,
+      type:
+        selectedTypeMeta !== null
+          ? {
+              text: selectedTypeMeta.displayName,
+              color: selectedTypeMeta.tagColor,
+            }
+          : undefined,
+      durationMinutes:
+        selectedTypeMeta?.fields?.duration === true && state.duration > 0
+          ? state.duration
+          : undefined,
+      tasks:
+        selectedTypeMeta?.hasTasks === true
+          ? state.tasks.map((task) => {
+              try {
+                return {
+                  ...YAML.parse(task.yamlContent),
+                  title: task.title,
+                };
+              } catch (e) {
+                return {
+                  title: task.title,
+                  description: 'An error occurred while parsing the task.',
+                };
+              }
+            })
+          : undefined,
+    },
+  };
+}
+
+const Creator = () => {
+  const { onNavigate } = useContext(AppContext);
+
+  const [state, setState] = useState<CreatorWizardState>({
+    type: null,
+    title: '',
+    description: '',
+    bundles: [],
+    url: '',
+    duration: 0,
+    tasks: [EMPTY_TASK],
+    introduction: '',
+    prerequisites: [],
+  });
+
+  const selectedType = useMemo(() => {
+    if (state.type === null) {
+      return null;
+    }
+
+    return { id: state.type, meta: itemKindMeta[state.type] };
+  }, [state.type]);
+
+  const quickStart = useMemo<QuickStart>(
+    () => makeDemoQuickStart(state),
+    [state]
+  );
+
+  const allQuickStarts = useMemo(() => [quickStart], [quickStart]);
+  const [activeQuickStart, setActiveQuickStart] = useState('');
+  const [quickStartStates, setQuickStartStates] = useState<AllQuickStartStates>(
+    {}
+  );
+
+  if (selectedType?.meta?.hasTasks !== true && activeQuickStart !== '') {
+    setActiveQuickStart('');
+  }
+
+  const parentContext = useContext(QuickStartContext);
+
+  const quickstartValues = useValuesForQuickStartContext({
+    allQuickStarts: allQuickStarts,
+    activeQuickStartID: activeQuickStart,
+    setActiveQuickStartID: (id) => setActiveQuickStart(id),
+    allQuickStartStates: quickStartStates,
+    setAllQuickStartStates: (states) => setQuickStartStates(states),
+    useQueryParams: false,
+    footer: parentContext.footer,
+    focusOnQuickStart: false,
+  });
+
+  useEffect(() => {
+    quickstartValues.setAllQuickStarts?.(allQuickStarts);
```

Comment:
And this was why `allQuickStarts` and `quickStart` were in `useMemo`. I'll look at just setting the state then, too.

Reply:
And again, the memo does nothing in this case. because the triggers are always object/array references because of the way the state management is setup.